File: sdk/basin-java/src/main/java/io/basin/sdk/ApiKeyIssued.java
package io.basin.sdk;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Response from {@code POST /auth/v1/api-keys}.
 *
 * <p>The {@link #secret} field is shown exactly once — store it immediately.
 */
public final class ApiKeyIssued {

    @JsonProperty("id")
    public final long id;

    @JsonProperty("name")
    public final String name;

    /** Raw API key secret — present only in this response, never retrievable again. */
    @JsonProperty("secret")
    public final String secret;

    @JsonProperty("created_at")
    public final String createdAt;

    public ApiKeyIssued(
            @JsonProperty("id") long id,
            @JsonProperty("name") String name,
            @JsonProperty("secret") String secret,
            @JsonProperty("created_at") String createdAt) {
        this.id = id;
        this.name = name;
        this.secret = secret;
        this.createdAt = createdAt;
    }
}

File: sdks/php/src/Basin/Types/ApiKeyIssued.php
<?php

declare(strict_types=1);

namespace Basin\Types;

/** POST /auth/v1/api-keys response — the secret is shown exactly once. */
final class ApiKeyIssued
{
    public function __construct(
        public readonly int $id,
        public readonly string $name,
        public readonly string $secret,
        public readonly string $createdAt,
    ) {
    }

    /** @param array<string,mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (int) $data['id'],
            name: (string) $data['name'],
            secret: (string) $data['secret'],
            createdAt: (string) $data['created_at'],
        );
    }
}
